Add Initializer into Container [MAILPOET-1823]

costasovo opened the pull request.

michelleshull commented:
I think some of the other hacking day commits may have knocked this out of sync, can you address the conflicts and send back? Thanks!

costasovo commented:
Rebased. I had to add one [commit](https://github.com/mailpoet/mailpoet/pull/1831/commits/3ac09279acbdd9aa62440b4183cbae7a33d7a173) since instantiating Mailer in constructor of a service we added to DI container in recent changes caused that it tried to load settings from DB before the database was instantiated.
So this needs additional review Can you take a look @wxa please?
I've also added an item for this to the DI refactor list in [the trello card](https://trello.com/c/VlleF1YT/58-di-refactor-checklist).
